### DISCUSSION TITLE
eslint 如何追加插件，例如import导入排序插件

### DISCUSSION THREAD

**Discussion by lynn1286** [umijs/umi#12077] [Q&A]
<img width="632" alt="image" src="https://github.com/umijs/umi/assets/40122311/fc8b4467-d837-42c6-a5f0-704704e09a06">
<img width="417" alt="image" src="https://github.com/umijs/umi/assets/40122311/705f7b0c-6170-498f-a6da-4d6050dee6c4">
这样配置以后，会与@umijs/max/eslint内置的产生冲突，然后就会有如下结果：

https://github.com/umijs/umi/assets/40122311/69ea76ac-d639-4ecd-aa9e-35194b5301db

**Comment by fz6m** [comment]
目测是和 prettier 插件里的 `prettier-plugin-organize-imports` 冲突了，umi 内置的 eslint 规则包含了一些很基础明显的语法错误检测，没有修复代码的功能，而格式化完全由 prettier 来做，如果你添加了 `eslint-plugin-import` 等带有自动修复代码功能的 eslint 插件，就需要面临一个传统经典问题：eslint 和 prettier 的格式化兼容。

这需要较多的配置来兼容这两种格式化行为变一致，因为 umi 本来就是一个强约束合集，所以不建议自己配置新的 eslint 插件。

如果你需要自己配置 eslint 功能，建议就不要 extends umi 的 eslint 配置了，完全从零开始配置 eslint config ，并且在配置时去兼容 prettier 和 eslint 的冲突问题（这个问题无论如何都存在，除非 eslint 没有代码格式化相关的约束，现在 umi extends 的就是这样的，不需要兼容他们俩），关于兼容的方法，需要你自己去学习。

注：在未来 eslint 马上要发布的下个大版本 v9 中，所有 eslint 的格式化代码规则都被去除了，意味着以后的大势是 eslint 并不做修复代码的功能，而是让 prettier 、biome 这些专门的代码格式化工具去做，当然社区也有一些人不喜欢这种变化所以有一些其他对策，此处介绍略。
